### PR TITLE
[#394]Compilation errors: Unused Variable and fread Error Handling

### PR DESCRIPTION
**Pull request by HazemRawi**

### CODE REVIEW

File: src/libpgmoneta/walfile/wal_reader.c [hunk 1]
```diff
@@ -129,7 +129,12 @@ read_all_page_headers(FILE* file, struct xlog_long_page_header_data* long_header
       fseek(file, page_number * long_header->xlp_xlog_blcksz, SEEK_SET);
       struct xlog_page_header_data* page_header = NULL;
       page_header = malloc(SIZE_OF_XLOG_SHORT_PHD);
-      fread(page_header, SIZE_OF_XLOG_SHORT_PHD, 1, file);
+      size_t bytes_read = fread(page_header, SIZE_OF_XLOG_SHORT_PHD, 1, file);
+      if (bytes_read < 0)
+      {
+         pgmoneta_log_error("Error:Failed to read the complete data\n");
```
Comment (jesperpedersen) [inline src/libpgmoneta/walfile/wal_reader.c:135]: You don't need the `\n`

Comment (jesperpedersen) [reply src/libpgmoneta/walfile/wal_reader.c:135]: Make an `error:` label and use a `goto`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/walfile/wal_reader.c [hunk 2]
```diff
@@ -138,6 +143,9 @@ read_all_page_headers(FILE* file, struct xlog_long_page_header_data* long_header
       pgmoneta_deque_add(wal_file->page_headers, NULL, (uintptr_t) page_header, ValueRef);
       page_number++;
    }
+error:
```
Comment (jesperpedersen) [inline src/libpgmoneta/walfile/wal_reader.c:146]: Doesn't look right - there should be a `return;` before that

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/walfile/wal_reader.c (hunk 2 above)
Comment (HazemRawi) [reply src/libpgmoneta/walfile/wal_reader.c:146]: before the `error:` label?

**Comment by jesperpedersen** [comment]
@sh-soltanpour Which compiler are you using ?

**Comment by HazemRawi** [comment]
**gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0**

**Comment by sh-soltanpour** [comment]
> @sh-soltanpour Which compiler are you using ?

mine is:
`cc (GCC) 14.1.1 20240701 (Red Hat 14.1.1-7)`

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**